gr-azure-software-radio/lib/difi_source_cpp_impl.cc

```diff
@@ -84,24 +112,95 @@ namespace gr {
                               gr_vector_const_void_star &input_items,
                               gr_vector_void_star &output_items)
     {
+      // check the file descriptor for incoming client connection
+      if(d_socket_type == SOCK_STREAM)
+      {
+        struct timeval tv;
+        tv.tv_sec = 0;
+        tv.tv_usec = 10000;
+
+        fd_set read_fds = d_rset;
+        int fd_max = *std::max_element(d_fds.begin(), d_fds.end());
+
+        if (select(fd_max+1, &read_fds, NULL, NULL, &tv) < 0)
+        {
+          GR_LOG_ERROR(this->d_logger, "Could not select server fd");
+          throw std::runtime_error("Could not select server fd");
+        }
+
+        for(auto fd : d_fds)
+        {
```

Comment:
this loop is hard to understand.

there is a return that stops the loop else it goes to completion, can we clean that up? (can stop the loop, save off the current fd (although I don't think that is needed since we don't use the fd in that case) and do buffer and send

also, we add to the collection we are iterating over here on line 147 d_fds. This is not good practice and can cause errors and odd bugs